src/Faithlife.Analyzers/AvailableWorkStateAnalyzer.cs

```diff
@@ -44,7 +44,7 @@ public override void Initialize(AnalysisContext context)
 	private static void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol iworkState, ISymbol workStateNone, ISymbol workStateToDo)
 	{
 		var propertyReferenceOperation = (IPropertyReferenceOperation) context.Operation;
-		if (!propertyReferenceOperation.Property.Equals(workStateNone) && !propertyReferenceOperation.Property.Equals(workStateToDo))
+		if (!SymbolEqualityComparer.Default.Equals(propertyReferenceOperation.Property, workStateNone) && !SymbolEqualityComparer.Default.Equals(propertyReferenceOperation.Property, workStateToDo))
```

Comment:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.symbolequalitycomparer.default?view=roslyn-dotnet-4.3.0 says "Compares two ISymbol instances based on the default comparison rules, equivalent to calling Equals(T)."

One normally assumes that `.Equals()` is an appropriate instance method to call to test equality. Is there documentation/justification for switching to an explicit equality comparer?

Reply:
"makes code look weirder", definitely true: https://twitter.com/kzu/status/1384195844308537356

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/3427#issuecomment-603859245

> When building against a version of roslyn that provides `SymbolEqualityComparer`, it should be used instead of `Equals` for comparing symbols.

RS1024 is raised and will enforce this guideline.

Reply:
Right, RS1024. Super annoying, indeed.